hudi-client/hudi-spark-client/src/test/java/org/apache/hudi/client/functional/TestHoodieBackedMetadata.java

```diff
@@ -507,6 +519,255 @@ public void testMetadataTableWithPendingCompaction(boolean simulateFailedCompact
     }
   }
 
+  /**
+   * Test arguments - Table type, populate meta fields, exclude key from payload.
+   */
+  public static List<Arguments> testMetadataRecordKeyExcludeFromPayloadArgs() {
+    return asList(
+        Arguments.of(COPY_ON_WRITE, true),
+        Arguments.of(COPY_ON_WRITE, false),
+        Arguments.of(MERGE_ON_READ, true),
+        Arguments.of(MERGE_ON_READ, false)
+    );
+  }
+
+  /**
```

Comment:
These tests are not specific to metadata table.

Populate meta fields should be tested via virtual key support.
key field should be tested via HFileWriter / HFileBlock tests.

Metadata Tests are time consuming so may be can reduce these.

Reply:
I initially had the testing at HFile writer and reader level, but it did not cover the compaction use case for the metadata table. The test here is more of everything combined and checks exactly what is needed for the metadata table records.